File: integrations/tool-tui/crates/check/playground/kotlin/Sample.kt
/**
 * Sample Kotlin file for dx-check testing.
 *
 * This file demonstrates Kotlin formatting and linting capabilities.
 */

package sample

/**
 * A generic stack implementation.
 */
class Stack<T> {
    private val data = mutableListOf<T>()

    fun push(value: T) {
        data.add(value)
    }

    fun pop(): T? {
        if (data.isEmpty()) return null
        return data.removeAt(data.size - 1)
    }

    fun peek(): T? {
        if (data.isEmpty()) return null
        return data.last()
    }

    fun isEmpty(): Boolean = data.isEmpty()

    fun size(): Int = data.size
}

/**
 * A simple calculator with operation history.
 */
class Calculator(initialValue: Double = 0.0) {
    var value: Double = initialValue
        private set

    private val history = mutableListOf<String>()

    fun add(x: Double): Calculator {
        value += x
        history.add("add($x)")
        return this
    }

    fun subtract(x: Double): Calculator {
        value -= x
        history.add("subtract($x)")
        return this
    }

    fun multiply(x: Double): Calculator {
        value *= x
        history.add("multiply($x)")
        return this
    }

    fun divide(x: Double): Calculator {
        require(x != 0.0) { "Cannot divide by zero" }
        value /= x
        history.add("divide($x)")
        return this
    }

    fun getHistory(): List<String> = history.toList()

    fun reset(): Calculator {
        value = 0.0
        history.clear()
        return this
    }
}

/**
 * Generate the first n Fibonacci numbers.
 */
fun fibonacci(n: Int): List<Long> {
    if (n <= 0) return emptyList()
    if (n == 1) return listOf(0L)

    val result = mutableListOf(0L, 1L)
    for (i in 2 until n) {
        result.add(result[i - 1] + result[i - 2])
    }
    return result
}

/**
 * Check if a string is a palindrome.
 */
fun isPalindrome(s: String): Boolean {
    val cleaned = s.filter { it.isLetterOrDigit() }.lowercase()
    return cleaned == cleaned.reversed()
}

/**
 * Extension function to format a list nicely.
 */
fun <T> List<T>.formatList(): String = joinToString(", ", "[", "]")

fun main() {
    // Test Stack
    val stack = Stack<Int>()
    stack.push(1)
    stack.push(2)
    stack.push(3)

    println("Stack size: ${stack.size()}")
    println("Top element: ${stack.peek()}")

    while (!stack.isEmpty()) {
        println("Popped: ${stack.pop()}")
    }

    // Test Fibonacci
    val fib = fibonacci(10)
    println("Fibonacci: ${fib.formatList()}")

    // Test palindrome
    val test = "A man a plan a canal Panama"
    val isPalin = if (isPalindrome(test)) "" else "not "
    println("\"$test\" is ${isPalin}a palindrome")

    // Test Calculator
    val calc = Calculator(10.0)
    calc.add(5.0).multiply(2.0).subtract(3.0)
    println("Calculator result: ${calc.value}")
    println("Calculator history: ${calc.getHistory().formatList()}")
}

File: integrations/tool-tui/crates/check/playground/php/sample.php
<?php

declare(strict_types=1);

/**
 * Sample PHP file for dx-check testing.
 *
 * This file demonstrates PHP formatting and linting capabilities.
 */

namespace Sample;

/**
 * A generic stack implementation.
 *
 * @template T
 */
class Stack
{
    /** @var array<int, T> */
    private array $data = [];

    /**
     * Push a value onto the stack.
     *
     * @param T $value
     */
    public function push(mixed $value): void
    {
        $this->data[] = $value;
    }

    /**
     * Pop a value from the stack.
     *
     * @return T|null
     */
    public function pop(): mixed
    {
        if (empty($this->data)) {
            return null;
        }
        return array_pop($this->data);
    }

    /**
     * Peek at the top value without removing it.
     *
     * @return T|null
     */
    public function peek(): mixed
    {
        if (empty($this->data)) {
            return null;
        }
        return $this->data[count($this->data) - 1];
    }

    /**
     * Check if the stack is empty.
     */
    public function isEmpty(): bool
    {
        return empty($this->data);
    }

    /**
     * Get the size of the stack.
     */
    public function size(): int
    {
        return count($this->data);
    }
}

/**
 * A simple calculator with operation history.
 */
class Calculator
{
    private float $value;
    /** @var array<int, string> */
    private array $history = [];

    public function __construct(float $initialValue = 0.0)
    {
        $this->value = $initialValue;
    }

    public function add(float $x): self
    {
        $this->value += $x;
        $this->history[] = "add({$x})";
        return $this;
    }

    public function subtract(float $x): self
    {
        $this->value -= $x;
        $this->history[] = "subtract({$x})";
        return $this;
    }

    public function multiply(float $x): self
    {
        $this->value *= $x;
        $this->history[] = "multiply({$x})";
        return $this;
    }

    public function divide(float $x): self
    {
        if ($x === 0.0) {
            throw new \InvalidArgumentException('Cannot divide by zero');
        }
        $this->value /= $x;
        $this->history[] = "divide({$x})";
        return $this;
    }

    public function getValue(): float
    {
        return $this->value;
    }

    /**
     * @return array<int, string>
     */
    public function getHistory(): array
    {
        return $this->history;
    }

    public function reset(): self
    {
        $this->value = 0.0;
        $this->history = [];
        return $this;
    }
}

/**
 * Generate the first n Fibonacci numbers.
 *
 * @return array<int, int>
 */
function fibonacci(int $n): array
{
    if ($n <= 0) {
        return [];
    }
    if ($n === 1) {
        return [0];
    }

    $result = [0, 1];
    for ($i = 2; $i < $n; $i++) {
        $result[] = $result[$i - 1] + $result[$i - 2];
    }
    return $result;
}

/**
 * Check if a string is a palindrome.
 */
function isPalindrome(string $s): bool
{
    $cleaned = preg_replace('/[^a-zA-Z0-9]/', '', $s);
    $cleaned = strtolower($cleaned ?? '');
    return $cleaned === strrev($cleaned);
}

// Main execution
function main(): void
{
    // Test Stack
    $stack = new Stack();
    $stack->push(1);
    $stack->push(2);
    $stack->push(3);

    echo "Stack size: " . $stack->size() . "\n";
    echo "Top element: " . $stack->peek() . "\n";

    while (!$stack->isEmpty()) {
        echo "Popped: " . $stack->pop() . "\n";
    }

    // Test Fibonacci
    $fib = fibonacci(10);
    echo "Fibonacci: " . implode(', ', $fib) . "\n";

    // Test palindrome
    $test = "A man a plan a canal Panama";
    $isPalin = isPalindrome($test) ? '' : 'not ';
    echo "\"{$test}\" is {$isPalin}a palindrome\n";

    // Test Calculator
    $calc = new Calculator(10);
    $calc->add(5)->multiply(2)->subtract(3);
    echo "Calculator result: " . $calc->getValue() . "\n";
    echo "Calculator history: " . implode(', ', $calc->getHistory()) . "\n";
}

main();
